src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <dwrite_3.h>
+#include <AppInstallerStrings.h>
+#include <winget/Fonts.h>
+
+namespace AppInstaller::Fonts
+{
+    namespace
+    {
+        std::vector<std::filesystem::path> GetFontFilePaths(const wil::com_ptr<IDWriteFontFace>& fontFace)
+        {
+            UINT32 fileCount = 0;
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, nullptr));
+
+            wil::com_ptr<IDWriteFontFile> fontFiles[8];
```

Comment:
1. static_assert that `sizeof(wil::com_ptr<IDWriteFontFile>) == sizeof(IDWriteFontFile*)`; means that it is a no-overhead smart pointer type and that you can use an array-like of them this way (regardless of the actual array-like type)
2. vector of `wil::com_ptr<IDWriteFontFile>`
3. resize to `fileCount`